addon/core/model/nodes/model-node.ts

```diff
@@ -143,6 +144,20 @@ export default abstract class ModelNode implements Walkable {
     return 1;
   }
 
+  get size(): number {
+    if (ModelNode.isModelText(this)) {
+      return this.content.length;
+    } else if (this.isLeaf) {
+      return 1;
+    } else if (ModelNode.isModelInlineComponent(this)) {
+      return 1;
+    } else if (ModelNode.isModelElement(this)) {
+      return this.children.reduce((prev, node) => prev + node.size, 0) + 2;
+    } else {
+      throw new NotImplementedError();
+    }
+  }
```

Comment:
If we introduce a size getter on modelnodes, we should probably do away with length (or rename it) as I can imagine it can lead to some confusion.

Reply:
its not the same thing though